testing/test_deep_learning_pipeline.py

```diff
@@ -75,12 +49,14 @@ def test_deep_learning_pipeline():
     # When
     p, outputs = p.fit_transform(data_inputs, expected_outputs)
 
+    metrics = p.apply('get_metrics')
+
     # Then
-    batch_mse_train = p.get_batch_metric_train('mse')
-    epoch_mse_train = p.get_epoch_metric_train('mse')
+    batch_mse_train = metrics['batch_metrics']['train']['mse']
+    epoch_mse_train = metrics['epoch_metrics']['train']['mse']
 
-    batch_mse_validation = p.get_batch_metric_validation('mse')
-    epoch_mse_validation = p.get_epoch_metric_validation('mse')
+    batch_mse_validation = metrics['batch_metrics']['validation']['mse']
+    epoch_mse_validation = metrics['epoch_metrics']['validation']['mse']
```

Comment:
here @guillaume-chevalier is this cool ?

Reply:
It looks fine, I hope this will stay even with my review of "apply" above. If this can't stay with my review of "apply", I'd be tempted to still do my review of apply... I'm not sure what to think anymore of how apply aggregates results.

Reply:
The aggregation of apply results is complex, I didn't understand on what dimension it was concatenating and when was it a dict or a list or something else, etc